beacon-chain/blockchain/service.go

```diff
@@ -226,6 +196,43 @@ func (c *ChainService) updateHead(slot uint64) {
 	c.candidateCrystallizedState = nilCrystallizedState
 }
 
+func (c *ChainService) getParentSlot(block *types.Block) (uint64, error) {
+	parentBlock, err := c.chain.getBlock(block.ParentHash())
+	if err != nil {
+		return 0, err
+	}
+
+	return parentBlock.SlotNumber(), nil
+}
+
+func (c *ChainService) doesParentExist(block *types.Block) bool {
+	parentExists, err := c.chain.hasBlock(block.ParentHash())
+	if err != nil {
+		log.Debugf("Could not check existence of parent hash: %v", err)
+		return false
+	}
+	if !parentExists {
+		log.Debugf("parent does not exist: %x", block.ParentHash())
+		return false
+	}
+
+	return true
+}
+
+func (c *ChainService) isPoWBlockValid(block *types.Block) bool {
```

Comment:
Is this function a stub? It's not checking anything and return `true` even when `web3Service` is `nil`. If it's a stub, can we please leave a comment or TODO?

Reply:
I think if the block doesn't exist, `fetcher.BlockByHash` returns an error. So this should really be called `doesPoWBlockExist`.

Reply:
I think this method can stay here, mostly because it doesn't fit anywhere else. I do agree that it's a little out of place though.